spec.emu

```diff
@@ -1005,42 +989,44 @@ contributors: Ron Buckton, Ecma International
           1. If IsCallable(_method_) is *false*, throw a *TypeError* exception.
         1. Return the DisposableResource Record { [[ResourceValue]]: _V_, [[Hint]]: _hint_, [[DisposeMethod]]: _method_ }.
       </emu-alg>
+      <emu-note>
+        Currently, the only allowed value for _hint_ is ~sync-dispose~.
+      </emu-note>
     </emu-clause>
 
     <emu-clause id="sec-getdisposemethod" type="abstract operation">
       <h1>
         GetDisposeMethod (
           _V_ : an Object,
-          _hint_ : either ~sync-dispose~ or ~async-dispose~,
+          _hint_ : ~sync-dispose~,
         )
       </h1>
       <dl class="header"></dl>
       <emu-alg>
-        1. If _hint_ is ~async-dispose~, then
-          1. Let _method_ be ? GetMethod(_V_, @@asyncDispose).
-          1. If _method_ is *undefined*, then
-            1. Set _method_ to ? GetMethod(_V_, @@dispose).
-        1. Else,
-          1. Let _method_ be ? GetMethod(_V_, @@dispose).
+        1. Let _method_ be ? GetMethod(_V_, @@dispose).
         1. Return _method_.
       </emu-alg>
+      <emu-note>
+        Currently, the only allowed value for _hint_ is ~sync-dispose~.
```

Comment:
In the AOs that ultimately would have used `_hint_`, I think it's clearer to have an `Assert: _hint_ is ~sync-dispose~.` instead of a note. In the callers of such AOs, it's fine to keep it as a note.